File: src/main/java/cn/chahuyun/economy/event/MessageEventListener.java
package cn.chahuyun.economy.event;

import cn.chahuyun.economy.utils.Log;
import kotlin.coroutines.CoroutineContext;
import net.mamoe.mirai.contact.BotIsBeingMutedException;
import net.mamoe.mirai.contact.MessageTooLargeException;
import net.mamoe.mirai.event.SimpleListenerHost;
import net.mamoe.mirai.event.events.EventCancelledException;
import org.jetbrains.annotations.NotNull;

/**
 * 说明
 *
 * @author Moyuyanli
 * @Description :消息检测
 * @Date 2022/7/9 18:11
 */
//@MessageComponent
public class MessageEventListener extends SimpleListenerHost {


    @Override
    public void handleException(@NotNull CoroutineContext context, @NotNull Throwable exception) {
        if (exception instanceof EventCancelledException) {
            Log.error("发送消息被取消:", exception);
        } else if (exception instanceof BotIsBeingMutedException) {
            Log.error("你的机器人被禁言:", exception);
        } else if (exception instanceof MessageTooLargeException) {
            Log.error("发送消息过长:", exception);
        } else if (exception instanceof IllegalArgumentException) {
            Log.error("发送消息为空:", exception);
        }

        // 处理事件处理时抛出的异常
        Log.error(exception.getCause());
    }



}

File: src/main/kotlin/cn/chahuyun/economy/event/MessageEventListener.kt
package cn.chahuyun.economy.event

import cn.chahuyun.economy.utils.Log
import net.mamoe.mirai.contact.BotIsBeingMutedException
import net.mamoe.mirai.contact.MessageTooLargeException
import net.mamoe.mirai.event.SimpleListenerHost
import net.mamoe.mirai.event.events.EventCancelledException
import kotlin.coroutines.CoroutineContext

/**
 * 消息检测
 */
@Deprecated("1.6.3")
class MessageEventListener : SimpleListenerHost() {

    override fun handleException(context: CoroutineContext, exception: Throwable) {
        when (exception) {
            is EventCancelledException -> Log.error("发送消息被取消:", exception)
            is BotIsBeingMutedException -> Log.error("你的机器人被禁言:", exception)
            is MessageTooLargeException -> Log.error("发送消息过长:", exception)
            is IllegalArgumentException -> Log.error("发送消息为空:", exception)
        }
        Log.error(exception.cause ?: exception)
    }
}
